go-controller/pkg/node/gateway_init.go

```diff
@@ -342,11 +342,21 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 	if portClaimWatcher != nil {
 		gw.portClaimWatcher = portClaimWatcher
 	}
-	initGw := func() error {
+
+	initGwFunc := func() error {
 		return gw.Init(n.watchFactory)
 	}
 
-	waiter.AddWait(gw.readyFunc, initGw)
+	readyGwFunc := func() (bool, error) {
+		controllerReady, err := n.isOVNControllerReady()
+		if err != nil || !controllerReady {
+			return false, err
+		}
+
+		return gw.readyFunc()
```

Comment:
you can call `waiter.AddWait` multiple times... might make more sense to add a second waitFunc rather than smushing this into the existing one?